### DISCUSSION TITLE
Branch "x" is not allowed to deploy to github-pages due to environment protection rules.

### DISCUSSION THREAD

**Discussion by nad-au** [community/community#39054] [Enterprise] [answered]
### Select Topic Area

General

### Body

I've created a new repository to host a private Pages site. We're on Enterprise. I'm using the recommended set-up for a Next Js site. Whether I deploy on main or any other branch I get this error.

![image](https://user-images.githubusercontent.com/6670753/201628494-19720e0b-6c99-49af-9bf8-acb4c971eed2.png)

I don't have any branch protection rules. I can push changes to any branch.

Please can you help me understand how to deploy a pages site? Or how to add or remove a protection rule so that I can deploy. Thanks.

**Comment by nad-au** [comment, marked as answer]
OK I've found the problem. Workflow creates an environment called `github-pages` which did indeed have a protection rule of branch `master`. Shortly after creating the repo I renamed default branch to `main`. So I had to update the env protection rule to match default branch and deployment worked.

**Reply by yoannchaudet** [reply]
Just wanted to add that this is a security mechanism. Any environment you deploy GitHub Pages to must be protected. By default we protect it with your repository's default branch.

**Reply by PetterGs** [reply]
Could this be elaborated on? Say I want to deploy from a branch called production. What do I need to do. I have no branch protection rules in my repo at all as far as I know.

**Reply by yoannchaudet** [reply]
We changed this behavior since my last message. As of today it is:

- If your site deploys with a branch source, the environment branch must be protected
- If your site deploys with a custom workflow (GitHub Actions source), the environment branch protection is optional (by default, we protect the `github-pages` environment with the repository's default branch, you can update this yourself or remove entirely the protection if you like)

**Comment by alberto56** [comment]
> So I had to update the env protection rule to match default branch and deployment worked.

Just to elaborate on this, this can be done by visiting settings for your repo, then "Environments", then "github-pages", then changing the allowed branch.

**Reply by elvisdragonmao** [reply]
Thanks bro, this really helps.

**Comment by evgfilim1** [comment]
Hello. I have the same problem, action says it cannot deploy from branch, but it's a tag.

![image](https://github.com/community/community/assets/17182159/28988f55-793a-4677-9555-5326af3b2f1f)

Protection rules: 
![image](https://github.com/community/community/assets/17182159/70924a82-2a04-4ab9-ab07-7e279d2a0a68)

**Reply by ragnarpa** [reply]
That started to happen out of the blue. GitHub, what is going on?

**Reply by jesshitchcock** [reply]
I am also experiencing this problem.

**Reply by jdogmcsteezy** [reply]
Me as well, but this behavior has been going on for some time for me.

**Reply by MathiasHaudgaard** [reply]
Having the same issue. 😞

**Reply by HenrySpartGlobal** [reply]
Had the same issue - I have heard you can contact GitHub support to revert the change for your account to the old version. But mid-September its going to be applied everywhere. GitHub engineering is working on getting something to fix these issues 

https://github.com/orgs/community/discussions/62991#discussioncomment-6844269

**Reply by davidkun** [reply]
Looks like this has been updated on GitHub's side. Now, when you enter the name pattern, you also get to choose between (1) `Ref type: branch` or (2) `Ref type: tag`. Make sure you've selected the correct type before entering the name pattern. 🙏

**Reply by EastSun5566** [reply]
Fixed by going to `Settings` → `Environments` → `github-pages` → `Selected branches and tags` → `Add deployment branch or tag rule`, then set the branch to `main` and the tags to `*`.

**Comment by albertogodar** [comment]
I am having the same issue, what's the status of this please?

**Comment by ppptrevoro** [comment]
Same issue here, can't deploy from a tag with branch protection rule starting yesterday

**Comment by fesaab** [comment]
Same issue here... It started to happen yesterday.

**Comment by dmikurube** [comment]
Same here for a tag. :(

However, this topic may be unrelated. I posted another discussion for the tag issue:
https://github.com/orgs/community/discussions/63890

**Comment by ap-vishal** [comment]
We are also having the same issue, it started happening on 10th August and it is probably due to this: https://github.blog/changelog/2023-08-10-actions-runs-triggered-from-tags-or-forks-with-the-same-name-as-a-protected-branch-will-now-be-blocked/

Right now, it breaks all of our projects.

**Comment by Tautorn** [comment]
Same issue here, I can't trigger build because github actions are returning `is not allowed to deploy to prod due to environment protection rules.`.  There's no protection rule for that case. All projects are broken.

**Comment by ivaivayl-flutterint** [comment]
<img width="782" alt="image" src="https://github.com/community/community/assets/111740817/d1a26c88-0ff7-4a72-82e6-91a73641e9f0">
This is currently happening within our project also. 
For example, previous version was deployed successfully through the workflow and it was "v0.10.2", now it's failing due to protection rules

**Comment by calebyouki** [comment]
it seems like GitHub doesn't know the difference between branches and tags

**Comment by mhw0** [comment]
Same issue:
![Screenshot 2023-08-19 at 10 51 50](https://github.com/community/community/assets/69183283/341b4ecf-52ad-4bb6-a338-fd8e4a7e1c12)

**Comment by cbarbian-sap** [comment]
Same here. We are deploying to `github-pages` from releases which have semver tags. In the past, we allowed a pattern `[0-9].*` (which is clearly not optimal), but even that is not working anymore now.

The only solution that works for us now is to allow all branches, which is obviously much worse. Is that really the plan?

**Comment by sunrawat** [comment]
instead of / use ./

**Comment by ashe-li** [comment]
https://github.com/withastro/docs/issues/1376

works for me

**Comment by rikhuijzer** [comment]
Another cause for this problem can be that the GitHub Pages source is set to "Deploy from branch" at 
```
https://github.com/<REPOSITORY_OWNER>/<REPOSITORY_NAME>/settings/pages
```
To fix it, set the source to "GitHub Actions". This fixed it for me on a repository which used to use `gh-pages` branch and switched to `actions/deploy-pages@v4`.

**Comment by geminixiang** [comment]
Settings → Environments → github-pages：and setup your branch name
